app/views/departments/index.html.erb

```diff
@@ -0,0 +1,80 @@
+<!--<h1>Listing Tickets</h1>-->
+
+<!--<table>-->
+<!--  <tr>-->
+<!--    <th>Subject</th>-->
+<!--    <th>Body</th>-->
+<!--    <th colspan="2"></th>-->
+<!--  </tr>-->
+
+<%# @tickets.each do |ticket| %>
+<!--    <tr>-->
+<!--      <td><%#= ticket.subject %></td>-->
+<!--      <td><%#= ticket.body %></td>-->
+<!--    </tr>-->
+<%# end %>
+<!--</table>-->
+
+<%#= button_to "Back", '/welcome', method: :get%>
+
+
+
+
+<h2>Listing Departments</h2>
+<ul class="nav nav-tabs">
+  <li class="text-right" role="presentation"><a href="/login">Logout</a></li>
+</ul>
+<% flash.each do |type, msg| %>
+  <div class="alert alert-info">
+    <%= msg %>
+  </div>
+<% end %>
+<%= form_tag('/search', :method => "get",
+             class: 'navbar-form navbar-left') do %>
+  <div class="input-group">
+    <%= search_field_tag :search, params[:search], placeholder: "Search", class: "form-control" %>
+    <div class="input-group-btn">
+      <div class="media-middle text-center">
+      <%= button_tag "", :class => 'btn btn-info glyphicon glyphicon-search',:name => nil%>
+    </div>
+    </div>
+  </div>
+<% end %>
+
+<%= form_tag 'admin-welcome', method: :get do %>
+
+  <div class="container">
+    <table class="table">
+      <thead>
+      <tr>
+        <th>Name</th>
+        <th>Tickets Generated</th>
+        <th colspan="2"></th>
+      </tr>
+      </thead>
+
+      <tbody>
+      <% @departments.each do |department| %>
+        <tr>
+          <td><%= department.name %></td>
```

Comment:
Instead of Ticket.where(:department_id => department.id)
Have a look at N+1 query problem in rails. Modify this accordingly